### ISSUE TITLE
Shouldn't vim-pyenv depends on vim's compile options

### ISSUE THREAD

**Issue by TKNGUE** [lambdalisue/vim-pyenv#19]
When I use vim compiled with option "+python3" to edit python file on python2 environment set by pyenv
or I use vim compiled with option "+python" to edit python file on python3 environment set by pyenv, 
vim-pyenv raise a error and say "vim-pyenv: Failed to activate "%s". Python version of the env is not supported in this Vim".

This behavior seems to be not favorable.
It force us to use vim +python3 enabled on python3 file and use vim enabled +python on python file. 

After just commenting out the code where to check vim's option in vim-env,
vim-pyenv still work well on my environment in the situation.

**Comment by lambdalisue** [comment]
Hum... While I use +python/+python3 Vim, I haven't really consider about the situation with +python/-python3 Vim or -python/+python3 Vim. So what kind of behaviour do you suggest me to implement?

**Comment by TKNGUE** [comment]
This problem is observed in OS X and some *nix.
My suggestion is very simple and is to remove some codes like this:
```diff
diff --git a/autoload/pyenv/pyenv.vim b/autoload/pyenv/pyenv.vim
index a81aefc..1255462 100644
--- a/autoload/pyenv/pyenv.vim
+++ b/autoload/pyenv/pyenv.vim
@@ -87,15 +87,7 @@ function! s:activate(name) abort " {{{
   let previous_PYENV_VERSION = $PYENV_VERSION
   let $PYENV_VERSION = a:name
   let external_python_major_version = pyenv#python#get_external_major_version()
-  if external_python_major_version == 2 && !has('python')
-    " cannot activate this python
-    let $PYENV_VERSION = previous_PYENV_VERSION
-    return 0
-  else if external_python_major_version == 3 && !has('python3')
-    " cannot activate this python
-    let $PYENV_VERSION = previous_PYENV_VERSION
-    return 0
-  endif
+
   let s:activated_name = a:name
   if g:pyenv#auto_create_ctags
     call s:create_ctags()
```
This code, at least, work well in my environment (In python3 environment of pyenv  by vim with +python).
That's because system-call in vim-pyenv calls python set by pyenv and jedi-vim calls python linked to vim when it is compiled.

**Comment by lambdalisue** [comment]
> That's because system-call in vim-pyenv calls python set by pyenv and jedi-vim calls python linked to vim when it is compiled

Well... probably I'm lost. I don't get why do you want to use vim-pyenv in such situation.

Calling `:PyenvActivate` assign `sys.path` of a python(pyenv) to a python(vim) if the major versions of both pythons are equal.
Assigning `sys.path` of python3 to python2 or opposite would trigger a lot of incompatible exceptions.
That's why vim-pyenv complains about the situation when the major version of python(pyenv) and python(vim) are different and Vim does not support the major version of python(pyenv).

Removing the code you showed allows vim-pyenv to go into the further procedure such as assigning `sys.path` so I don't really think that what you want.

You may miss [`g:pyenv#auto_activate`](https://github.com/lambdalisue/vim-pyenv/blob/master/doc/vim-pyenv.txt#L180) option or did I get the wrong idea?

**Comment by TKNGUE** [comment]
Most I expect is to make python(pyenv) searchable by jedi-vim.

I want you to know that there are cases where we can't enable both +python/+python3 options of vim.
I'm pretty sure that such cases are majority of vim-user(enabling only either +python/+python3).
What I want to emphasize is that coding environment(vim's python) and application environment(pyenv's python) must be isolated(No wonder that people develop Python2.7 applications on vim(+python3) ).

>Assigning sys.path of python3 to python2 or opposite would trigger a lot of incompatible exceptions.
That's why vim-pyenv complains about the situation when the major version of python(pyenv) and python(vim) are different and Vim does not support the major version of python(pyenv).

Indeed, Adding different versions sys.path of pyenv into python(vim)'s sys path allow vim to import, but side-effect is little and risk is also controllable by changing sys.path order's.
Unless we deliberately try to use libraries of python(pyenv) from python(vim) (if such case, it's plugin's fault) exceptions will not happen.
So far, such exceptions haven't been caused on my environment(OS X, jedi-vim + pyenv and other python-implemented vim plugins)

According to jedi-vim implementations,   we can add paths search by setting VIRTUAL_ENV.  It may be pretty good rather than just adding sys.path.

**Comment by lambdalisue** [comment]
> Indeed, Adding different versions sys.path of pyenv into python(vim)'s sys path allow vim to import, but side-effect is little and risk is also controllable by changing sys.path order's.
> Unless we deliberately try to use libraries of python(pyenv) from python(vim) (if such case, it's plugin's fault) exceptions will not happen.

Well, vim-pyenv designed to **use** modules of python(pyenv) from python(vim) so that e.g. jedi-vim can construct its candidates by importing the module rather than parsing (AFIK, jedi-vim requires to import modules to construct candidates). In such case, if the module use a old-style (e.g. `print 'msg'`) and Vim is compiled with Python3.x or the module use a latest-style (e.g. `print('msg')`) and Vim is compiled with Python2.6 or earlier, tons of RuntimeError would be caused.

Recently most of fundamental libraries use a latest-styles (`print('msg')` or `except Exception as e`) and majority users use Vim compiled with python2.7 and/or python3.x so such RuntimeError would not be caused (while python2.7 is compatible with python3.x styles). But there are some libraries which still use old-style and some users which need to use a Vim compiled with python2.6 or earlier.

> According to jedi-vim implementations, we can add paths searched by setting VIRTUAL_ENV. It may be pretty good rather than just adding sys.path.

The main motivation of the development of vim-pyenv is to support jedi-vim but the purpose of vim-pyenv is to allow switching environments of pyenv in Vim's live session. So I don't want to rely on the implementation of jedi-vim.

Additionally, `$VIRTUAL_ENV` is an environment variable which [`virtualenv`](https://virtualenv.pypa.io/en/latest/index.html) expose ([ref](https://github.com/pypa/virtualenv/blob/develop/virtualenv_embedded/activate.sh#L44)) so actually it is not an original implementation of jedi-vim. It is a feature of virtualenv so vim-pyenv **should not** overwrite `$VIRTUAL_ENV`.

**TL;DR:** As I said, as long as the Vim is compiled with python2.7, numerous RuntimeError of python2/python3 incompatibility would possibly be ignored. Additionally now I understand that users would like to allow e.g. jedi-vim to complete function names or whatever of python3 libraries in Vim compiled with python2. **So how about adding an option (say, `g:pyenv#ignore_external_version_check`) to ignore version check and force vim-pyenv to overwrite `sys.path`? Do you think that solve the issue?**

**Comment by TKNGUE** [comment]
I mistakenly understand jedi-vim's behavior, which parse python files by itself.
To add option is compromising idea. 

Okay, I'll send pull request.
